### PR TITLE
Move `updateExtensionByColonyAndHash` lamda to block ingestor

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Moved `updatedExtensionByColonyAndHash` lambda function logic to the block-ingestor as it's only used here.

Resolves [colonyCDapp#339](https://github.com/JoinColony/colonyCDapp/issues/339)

### CODE REVIEW

File: src/graphql/queries/extensions.graphql [hunk 1]
```diff
@@ -36,3 +36,19 @@ query ListExtensions($hash: String!, $nextToken: String) {
     }
   }
 }
+
+query GetColonyExtensionByHashAndColony($extensionHash: String!, $colonyAddress: ID!) {
+  listColonyExtensions(filter: {hash: {eq: $extensionHash}, colonyId: {eq: $colonyAddress}}) {
```
Comment (willm30) [inline src/graphql/queries/extensions.graphql:41]: There is a `getExtensionByColonyAndHash` query that you can call directly.

File: src/handlers/extensions/extensionDeprecated.ts [hunk 2]
```diff
@@ -18,14 +21,28 @@ export default async (event: ContractEvent): Promise<void> => {
...
+  const extensionId = data?.listColonyExtensions?.items[0]?.id;
+
+  if (extensionId) {
+    await mutate<
+      UpdateColonyExtensionByAddressMutation,
+      UpdateColonyExtensionByAddressMutationVariables
+    >(UpdateColonyExtensionByAddressDocument, {
+      input: {
+        id: extensionId,
+        colonyId: colony,
```
Comment (willm30) [inline src/handlers/extensions/extensionDeprecated.ts:42]: Once you have the extension id you don't need to pass `colonyId` or `hash` to the mutation

File: src/handlers/extensions/extensionUpgraded.ts [hunk 3]
```diff
@@ -20,14 +23,28 @@ export default async (event: ContractEvent): Promise<void> => {
...
+  if (extensionId) {
+    await mutate<
+      UpdateColonyExtensionByAddressMutation,
+      UpdateColonyExtensionByAddressMutationVariables
+    >(UpdateColonyExtensionByAddressDocument, {
+      input: {
+        id: extensionId,
+        colonyId: colony,
+        hash: extensionHash,
+        version: convertedVersion,
```
Comment (willm30) [inline src/handlers/extensions/extensionUpgraded.ts:46]: ditto

File: src/utils/extensions/writeExtension.ts [hunk 4]
```diff
@@ -85,41 +88,80 @@ export const writeExtensionFromEvent = async (
...
+        id: extensionId,
+        isDeleted: true,
+      },
     });
-  } else {
+  }
+};
+
+const createOrUpdateColonyExtension = async (
+  input: CreateColonyExtensionMutationVariables['input'],
```
Comment (willm30) [inline src/utils/extensions/writeExtension.ts:126]: There's a type called: `CreateColonyExtensionInput`

File: src/utils/extensions/writeExtension.ts [hunk 5]
```diff
@@ -85,41 +88,80 @@ export const writeExtensionFromEvent = async (
...
       CreateColonyExtensionMutationVariables
     >(CreateColonyExtensionDocument, {
       input: {
-        id: extensionAddress,
         ...input,
+        id: extensionAddress,
+      },
+    });
+  } else {
+    // Otherwise, update the exising extension
```
Comment (willm30) [inline src/utils/extensions/writeExtension.ts:153]: typo

**Review by willm30 — APPROVED** [review]
LGTM, only thing I would recommend is swapping out the `list` query for the `getExtensionByColonyAndHash` query

**Review by jakubcolony — APPROVED** [review]
LGTM 👍

**Pull request merged**